pandas/_libs/parsers.pyx

```diff
@@ -1195,7 +1196,10 @@ cdef class TextReader:
                 na_count = 0
 
             if result is not None and dtype != 'int64':
-                result = result.astype(dtype)
+                if is_extension_array_dtype(dtype):
+                    result = result.astype(dtype.numpy_dtype)
```

Comment:
you will need to use 
```
result = dtype.construct_array_type()._from_sequence(result)
```

Reply:
may need a try/except around this though

Reply:
We would probably want to raise if the result can't be constructed, right?

Reply:
Yeah I would think so. That would be the defensive thing to do at least. Is that typically what is done in other places?

Reply:
Depends, but in this case I think raising is appropriate.

Reply:
Raising here would actually be a breaking change I think. It broke several tests at least. 

If a user already has an EA class that adheres to the current interface it would all the sudden not work until they defined the new `_from_sequence_of_strings` method. We could try to use the new method and then fallback to the old behavior and warn? 

Not sure how you want to handle this, but I'm pretty sure that it will be a breaking change if we just raised if we couldn't use the new method.